### ISSUE TITLE
Easier MentionObject API

### ISSUE THREAD

**Issue by falcaopetri** [jheddings/notional#47]
### Description ###

I was trying to create a page titled "@ Today". This requires a `types.MentionObject(types.MentionDate()))` object. The best snippets I could come up with were:

1. `types.MentionObject.parse_obj`
```python
from datetime import datetime
today = datetime.now().strftime('%Y-%m-%d')

mention_data = {
    "type": "mention",
    "plain_text": f"{today} → ",
    "mention": {"type": "date", "date": {"start": today, "end": None}},
}
types.Title(title=[types.MentionObject.parse_obj(mention_data)])
```
based on https://github.com/jheddings/notional/blob/391d16fbf53782599d7f4f13fe92a49fdafbaef0/tests/test_types.py#L657

2. `types.MentionObject(types.MentionDate(types.DateRange))`

```python
from datetime import datetime
today = datetime.now().strftime('%Y-%m-%d')

types.Title(title=[types.MentionObject(plain_text=f"{today} → ", mention=types.MentionDate(date=types.DateRange(start=today)))]),
```
based on 
```python
page = notion.pages.retrieve(PAGE_ID) # a page with title inserted in Notion
page["title"]
```

Moreover, `Title` does not seem to allow composition with mentions since it assumes a `TextObject`:
https://github.com/jheddings/notional/blob/d96cefbecfbe406b820fbec593e36ea3a886099a/notional/types.py#L266

### Questions ###

- Is currently a better approach for creating a mention?
- If not, could there be a better API for creating a `MentionObject`? For example, by automagically wrapping a `MentionData` inside a `MentionObject`, or having an easier way to build `mention_data`?

**Comment by jheddings** [comment]
@falcaopetri Finally getting some time to look into this...  I think I have an idea, but would appreciate your feedback.  Would you mind taking a look at [this branch](https://github.com/jheddings/notional/tree/easier_mention) to see if it would be better for you?  In particular, see if the new tests are helpful in creating the various mention objects.

**Comment by falcaopetri** [comment]
Nice work on improving the API @jheddings! I was able to test the following:

```python
from datetime import date, timedelta

tomm = date.today() + timedelta(days=1)
page = notion.pages.retrieve(EMPTY_NOTION_PAGE)

at = types.MentionDate["Mention Tomorrow", tomm]
text = types.TextObject["Entry from "]

# usage scenario 1
notion.pages.update(page, title=types.Title(title=[at]))
# usage scenario 2
notion.pages.update(page, title=types.Title(title=[text, at]))
```

Both tests worked and provided the intended results with a much cleaner code.

**Comment by jheddings** [comment]
Thanks for the feedback and example!  I'll get this into the next release.

I also see your point on the `types.Title` object, which should support composing a `Title` from an array of `RichTextObject`'s.  I'll give that a look tonight and hopefully have an update for you to try.

**Comment by jheddings** [comment]
@falcaopetri I took another pass at this after poking around a bit.  It appears that the `plain_text` for "mentions" is not very relevant, so I've removed it from the compose interface for the time being.

Additionally, this was a good opportunity to improve composing `Title` (and other text blocks) based on your feedback.  Feel free to take a look at the PR (#94) and provide feedback.

Thanks again!
